connectors/sources/sharepoint_online.py

```diff
@@ -1426,7 +1452,36 @@ async def process_user(user):
             if user_doc:
                 yield user_doc
 
-    async def _drive_items_batch_with_permissions(self, drive_id, drive_items_batch):
+    async def site_group_users(self, site_web_url, site_group_id):
+        """
+        Fetches the users of a given site group. Checks in-memory cache before making an API call.
+
+        Parameters:
+        - site_web_url (str): The URL of the site collection.
+        - site_group_id (int): The ID of the site group.
+
+        Returns:
+        - list: List of users for the given site group.
+        """
+
+        # Check cache first
+        if site_group_id in self.site_group_cache:
```

Comment:
is this `site_group_id` one that's globally unique? It looks like it's just coming off of `user.get("id")`, which I thought would be values like `1`, `2`, etc.

Reply:
I somehow wanted to combine `site_web_url, site_group_id` into globally unique id for caching but I totally forgot to do it...

Reply:
Should be resolved now